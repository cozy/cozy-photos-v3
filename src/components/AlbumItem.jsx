import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router-dom'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true,
      isImageLoading: true
    }

    this.handleImageLoaded = this.handleImageLoaded.bind(this)
    this.fetchPhoto = this.fetchPhoto.bind(this)

    if (props.album && props.album.photosIds.length) {
      this.fetchPhoto(props.album.photosIds[0])
    } else {
      this.state = {
        url: '',
        isLoading: false
      }
    }
  }

  fetchPhoto (photoId) {
    getPhotoLink(photoId)
      .then(link => this.setState({
        url: link,
        isLoading: false
      }))
  }

  handleImageLoaded () {
    this.setState({ isImageLoading: false })
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url, isImageLoading } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              onLoad={this.handleImageLoaded}
              style={isImageLoading ? 'display:none' : ''}
              alt={`${album.name} album cover`}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
