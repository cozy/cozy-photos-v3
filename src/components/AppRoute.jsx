import React from 'react'
import { Route, Redirect } from 'react-router'

import Layout from './Layout'
import Timeline from '../containers/Timeline'
import AlbumsView from '../containers/AlbumsView'
import AlbumPhotos from '../containers/AlbumPhotos'
import Viewer from '../components/Viewer'
import { PhotosPicker } from '../ducks/albums'

export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={Layout}>
    <Route path='photos' component={Timeline}>
      <Route path=':photoId' component={Viewer} />
    </Route>
    <Route path='albums' component={AlbumsView}>
      <Route path='new' component={PhotosPicker} />
      <Route path=':albumId' component={AlbumPhotos}>
        <Route path='edit' component={PhotosPicker} />
        <Route path=':photoId' component={Viewer} />
      </Route>
    </Route>
    <Redirect from='/*' to='photos' />
  </Route>
)

export default AppRoute
