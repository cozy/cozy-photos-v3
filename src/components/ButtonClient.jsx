import styles from '../styles/buttonclient'

import { translate } from '../lib/I18n'

import React from 'react'

const ButtonClient = ({ t }) => (
  <a href={t('Nav.link-client')} target='_blank' className={styles['coz-btn-client']}><span>{t('Nav.btn-client')}</span></a>
)

export default translate()(ButtonClient)
