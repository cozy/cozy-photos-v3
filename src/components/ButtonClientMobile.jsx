/* global localStorage */

import styles from '../styles/buttonclient'

import { translate } from '../lib/I18n'

import React, { Component } from 'react'

import classNames from 'classnames'

class ButtonClientMobile extends Component {
  constructor (props) {
    super(props)
    this.state = {
      seen: JSON.parse(localStorage.getItem('app_ad')) || false
    }
  }

  read () {
    localStorage.setItem('app_ad', JSON.stringify(true))
    this.setState({seen: true})
  }

  render ({ t }, { seen }) {
    if (seen) return null

    return (
      <div className={styles['coz-banner-client']}>
        <a href={t('Nav.link-client')} target='_blank' className={styles['coz-btn-clientMobile']} onClick={e => this.read(e)} ><span>{t('Nav.btn-client-mobile')}</span></a>
        <button className={classNames('coz-btn', styles['coz-btn--close'])} onClick={e => this.read()} />
      </div>
    )
  }
}

export default translate()(ButtonClientMobile)
