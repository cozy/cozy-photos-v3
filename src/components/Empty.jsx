import styles from '../styles/emptyAndError'

import React from 'react'
import { translate } from '../lib/I18n'

export const Empty = ({ t, emptyType }) => {
  return (
    <div className={styles['pho-empty']}>
      <h2>{ t(`Empty.${emptyType}_title`) }</h2>
      <p>{ t(`Empty.${emptyType}_text`)}</p>
    </div>
  )
}

const TranslatedEmpty = translate()(Empty)

export const withEmpty = (isEmpty, type, BaseComponent) => props =>
  isEmpty(props)
    ? <TranslatedEmpty emptyType={type} />
    : <BaseComponent {...props} />

export default TranslatedEmpty
