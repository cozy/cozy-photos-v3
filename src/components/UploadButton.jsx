import React from 'react'
import { translate } from '../lib/I18n'

const styles = {
  parent: {
    position: 'relative',
    overflow: 'hidden'
  },
  input: {
    position: 'absolute',
    top: 0,
    left: 0,
    opacity: 0,
    width: '100%',
    height: '100%',
    zIndex: 1
  }
}

export const UploadButton = ({ t, disabled, onUpload }) => (
  <label
    role='button'
    disabled={disabled}
    className='coz-btn coz-btn--regular coz-btn--upload'
    style={styles.parent}
  >
    { t('Toolbar.photo_upload') }
    <input
      type='file'
      accept='image/*'
      multiple
      disabled={disabled}
      style={styles.input}
      onChange={e => {
        // e.target.files is an array-like, transform it to Array instance
        const photosArray = Array.from(e.target.files)
        onUpload(photosArray)
      }} />
  </label>
)

export default translate()(UploadButton)
