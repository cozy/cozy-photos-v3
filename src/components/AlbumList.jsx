// import styles from '../styles/albumList'

import React from 'react'
import { translate } from '../lib/I18n'

import Empty from '../components/Empty'
import Topbar from '../components/Topbar'

export const PhotosList = ({ t }) => {
  return (
    <div>
      <Topbar viewName='albums' />
      <div role='contentinfo'>
        <Empty emptyType='albums' />
      </div>
    </div>
  )
}

export default translate()(PhotosList)
