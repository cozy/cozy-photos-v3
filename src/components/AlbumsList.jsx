import styles from '../styles/albumsList'

import React from 'react'

import { withEmpty } from '../components/Empty'
import AlbumItem from '../components/AlbumItem'
import NewAlbum from '../components/NewAlbum'

const DumbAlbumsList = props => (
  <div className={styles['pho-album-list']}>
    {props.albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={props.onServerError} />)}
    <NewAlbum />
  </div>
)

const AlbumsList = withEmpty(props => props.albums.length === 0, 'albums', DumbAlbumsList)

export default AlbumsList
