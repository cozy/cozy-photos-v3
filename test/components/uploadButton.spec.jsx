'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT } from '../lib/I18n'
import { UploadButton } from '../../src/components/UploadButton'

describe('UploadButton component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <UploadButton t={mockT} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
