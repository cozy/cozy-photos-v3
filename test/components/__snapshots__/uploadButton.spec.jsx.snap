// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton component should be rendered correctly 1`] = `
<label
  className=" coz-btn coz-btn--regular coz-btn--upload"
  disabled={undefined}
  role="button"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <input
    accept="image/*"
    disabled={undefined}
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
    type="file"
  />
</label>
`;

exports[`UploadButton component should be rendered correctly if type is menu-item 1`] = `
<label
  className=" coz-link--upload"
  disabled={undefined}
  role="button"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <input
    accept="image/*"
    disabled={undefined}
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
    type="file"
  />
</label>
`;

exports[`UploadButton component should handle onChange event 1`] = `
<label
  className=" coz-btn coz-btn--regular coz-btn--upload"
  disabled={undefined}
  role="button"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <input
    accept="image/*"
    disabled={undefined}
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
    type="file"
  />
</label>
`;
